internal/healthcheck/healthcheck.go

```diff
@@ -90,8 +89,24 @@ func (c *Check) HandleReadiness(w http.ResponseWriter, _ *http.Request) {
 		return
 	}
 
-	err := c.proxy.CheckConnections(ctx)
-	if err != nil && !ready(err, c.minReady) {
+	var minReady int
+	q := req.URL.Query()
+	if v := q.Get("min-ready"); v != "" {
+		n, err := strconv.Atoi(v)
+		// If the query value parsed correctly, set it to minReady. Otherwise,
+		// ignore it.
+		if err == nil {
+			minReady = n
+		}
+	}
+
+	n, err := c.proxy.CheckConnections(ctx)
+	if minReady > n {
```

Comment:
Between this line and ready I feel like I'm confused. Is n the number of connections or the number of instances?

Reply:
I've cleaned this up to make it more clear.